File: sgi-webapp/src/app/core/models/eti/tipo-evaluacion.ts
import { marker } from '@biesbjerg/ngx-translate-extract-marker';

export enum TIPO_EVALUACION {
  RETROSPECTIVA = 1,
  MEMORIA = 2,
  SEGUIMIENTO_ANUAL = 3,
  SEGUIMIENTO_FINAL = 4
}

export const TIPO_EVALUACION_MAP: Map<TIPO_EVALUACION, string> = new Map([
  [TIPO_EVALUACION.RETROSPECTIVA, marker(`eti.tipo-evaluacion.RETROSPECTIVA`)],
  [TIPO_EVALUACION.MEMORIA, marker(`eti.tipo-evaluacion.MEMORIA`)],
  [TIPO_EVALUACION.SEGUIMIENTO_ANUAL, marker(`eti.tipo-evaluacion.SEGUIMIENTO_ANUAL`)],
  [TIPO_EVALUACION.SEGUIMIENTO_FINAL, marker(`eti.tipo-evaluacion.SEGUIMIENTO_FINAL`)]
]);

export class TipoEvaluacion {

  /** ID */
  id: number;

  /** Nombre */
  nombre: string;

  /** Activo */
  activo: boolean;

  constructor() {
    this.id = null;
    this.nombre = null;
    this.activo = true;
  }

}

File: sgi-rep-service/src/main/java/org/crue/hercules/sgi/rep/enums/TipoEvaluacion.java
package org.crue.hercules.sgi.rep.enums;

import org.crue.hercules.sgi.rep.report.SgiReportHelper;

public enum TipoEvaluacion {
  /** Retrospectiva <code>1L</code> */
  RETROSPECTIVA(1L, "enum.tipo-evaluacion.RETROSPECTIVA"),
  /** Memoria <code>2L</code> */
  MEMORIA(2L, "enum.tipo-evaluacion.MEMORIA"),
  /** Seguimiento anual <code>3L</code> */
  SEGUIMIENTO_ANUAL(3L, "enum.tipo-evaluacion.SEGUIMIENTO_ANUAL"),
  /** Seguimiento final <code>4L</code> */
  SEGUIMIENTO_FINAL(4L, "enum.tipo-evaluacion.SEGUIMIENTO_FINAL");

  private final Long id;
  private final String i18nMessage;

  private TipoEvaluacion(Long id, String i18nMessage) {
    this.id = id;
    this.i18nMessage = i18nMessage;
  }

  public Long getId() {
    return this.id;
  }

  @Override
  public String toString() {
    return SgiReportHelper.getMessage(this.i18nMessage);
  }

  public static TipoEvaluacion fromCode(final Long id) {
    for (TipoEvaluacion tipo : TipoEvaluacion.values()) {
      if (tipo.getId().equals(id)) {
        return tipo;
      }
    }
    return null;
  }
}
